src/solvers/sat/satcheck_minisat2.cpp

```diff
@@ -110,213 +95,174 @@ void satcheck_minisat2_baset<T>::lcnf(const bvt &bv)
 
   // MiniSat2 kills the model in case of UNSAT
-  solver->model.growTo(v+1);
+  solver_->model.growTo(v+1);
   value^=sign;
-  solver->model[v]=Minisat::lbool(value);
+  solver_->model[v]=Minisat::lbool(value);
 }
 
-template<typename T>
-satcheck_minisat2_baset<T>::satcheck_minisat2_baset(T *_solver):
-  solver(_solver), time_limit_seconds(0)
+satcheck_minisat2_baset::satcheck_minisat2_baset(
+  std::unique_ptr<Minisat::Solver> solver,
+  const double random_seed,
+  const double random_var_freq):
+  solver_(std::move(solver))
 {
+  solver_->random_seed=random_seed;
+  solver_->random_var_freq=random_var_freq;
 }
 
-template<>
-satcheck_minisat2_baset<Minisat::Solver>::~satcheck_minisat2_baset()
-{
-  delete solver;
-}
-
-template<>
-satcheck_minisat2_baset<Minisat::SimpSolver>::~satcheck_minisat2_baset()
-{
-  delete solver;
-}
+satcheck_minisat2_baset::~satcheck_minisat2_baset() = default;
 
-template<typename T>
-bool satcheck_minisat2_baset<T>::is_in_conflict(literalt a) const
+bool satcheck_minisat2_baset::is_in_conflict(literalt a) const
 {
-  int v=a.var_no();
+  const int v=a.var_no();
 
-  for(int i=0; i<solver->conflict.size(); i++)
-    if(var(solver->conflict[i])==v)
+  for(int i=0; i<solver_->conflict.size(); i++)
+    if(var(solver_->conflict[i])==v)
       return true;
 
   return false;
 }
 
-template<typename T>
-void satcheck_minisat2_baset<T>::set_assumptions(const bvt &bv)
+void satcheck_minisat2_baset::set_assumptions(const bvt &bv)
 {
-  assumptions=bv;
+  assumptions_=bv;
 
-  forall_literals(it, assumptions)
+  forall_literals(it, assumptions_)
     if(it->is_true())
     {
-      assumptions.clear();
+      assumptions_.clear();
       break;
     }
 }
 
-satcheck_minisat_no_simplifiert::satcheck_minisat_no_simplifiert():
-  satcheck_minisat2_baset<Minisat::Solver>(new Minisat::Solver)
-{
-}
-
-satcheck_minisat_simplifiert::satcheck_minisat_simplifiert():
-  satcheck_minisat2_baset<Minisat::SimpSolver>(new Minisat::SimpSolver)
-{
-}
+satcheck_minisat_no_simplifiert::satcheck_minisat_no_simplifiert(
+  const double random_seed,
+  const double random_var_freq):
+  satcheck_minisat2_baset(
+    util_make_unique<Minisat::Solver>(),
+    random_seed,
+    random_var_freq)
+{ }
+
+satcheck_minisat_simplifiert::satcheck_minisat_simplifiert(
+  const double random_seed,
+  const double random_var_freq):
+  satcheck_minisat2_baset(
+    util_make_unique<Minisat::SimpSolver>(),
+    random_seed,
+    random_var_freq)
+{ }
 
 void satcheck_minisat_simplifiert::set_frozen(literalt a)
 {
   if(!a.is_constant())
   {
     add_variables();
-    solver->setFrozen(a.var_no(), true);
+    static_cast<Minisat::SimpSolver*>(solver_.get())->setFrozen(a.var_no(), true);
   }
 }
 
 bool satcheck_minisat_simplifiert::is_eliminated(literalt a) const
 {
   assert(!a.is_constant());
 
-  return solver->isEliminated(a.var_no());
+  return static_cast<const Minisat::SimpSolver*>(solver_.get())->
+    isEliminated(a.var_no());
 }
```

Comment:
There seem to be an awful lot of "refactorings" here which are not related to the actual PR.  Please do not do this.  Refactorings should be a separate commit.

Reply:
It should be said that @LAJW has done this in a pretty clean series of commits. I do not know what the exact request in TG-925 is; if it is what the PR subject says, then surely refactorings have no place in here. If TG-925 requested refactoring, then these should be better be considered for other solvers as well.